internal/cutil/ptrguard_test.go

```diff
@@ -0,0 +1,66 @@
+package cutil
+
+import (
+	"math/rand"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPtrGuard(t *testing.T) {
+	t.Run("storeAndRelease", func(t *testing.T) {
+		s := "string"
+		goPtr := (unsafe.Pointer)(&s)
+		cPtr := Malloc(PtrSize)
+		defer Free(cPtr)
+		pg := NewPtrGuard(cPtr, goPtr)
+		assert.Equal(t, *(*unsafe.Pointer)(cPtr), goPtr)
+		pg.Release()
+		assert.Zero(t, *(*unsafe.Pointer)(cPtr))
+	})
+
+	t.Run("multiRelease", func(t *testing.T) {
+		s := "string"
+		goPtr := (unsafe.Pointer)(&s)
+		cPtr := Malloc(PtrSize)
+		defer Free(cPtr)
+		pg := NewPtrGuard(cPtr, goPtr)
+		assert.Equal(t, *(*unsafe.Pointer)(cPtr), goPtr)
+		pg.Release()
+		pg.Release()
+		pg.Release()
+		pg.Release()
+		assert.Zero(t, *(*unsafe.Pointer)(cPtr))
+	})
+
+	t.Run("stressTest", func(t *testing.T) {
+		const N = 20000 // assert that no threads are created (default limit 10000)
```

Comment:
I assume that the 20000 size is would not be reached, and when > 10000 there will be a panic? A comment what type of 'assert' you do here would be nice.

Reply:
Yes, if the PtrGuards would each create a thread (as they did before), it would panic with more than 10000, because that is the default thread limit for the Go runtime. I'll improve the comment.